scripts/reports/ssr-progress/ssr-progress.mjs

```diff
@@ -0,0 +1,97 @@
+import fs from 'fs';
+import {execute} from '../../exec.mjs';
+
+async function buildHeadless() {
+  await execute('npx', ['nx', 'run', 'headless:build']);
+}
+
+function getParsedDoc() {
+  const rawdata = fs.readFileSync(
+    'packages/headless/dist/parsed_doc.json',
+    'utf8'
+  );
+  const rawSSRdata = fs.readFileSync(
+    'packages/headless/dist/ssr_parsed_doc.json',
+    'utf8'
+  );
+  const parsed_doc = JSON.parse(rawdata);
+  const ssr_parsed_doc = JSON.parse(rawSSRdata);
+
+  return {parsed_doc, ssr_parsed_doc};
+}
+
+function removeControllersWithNoSSRSupport(useCase) {
+  if (useCase.name === 'search') {
+    return useCase.controllers.filter((controller) => {
+      return controller.initializer.name !== 'buildUrlManager';
+    });
+  }
+}
```

Comment:
Here is the function for removing certain controllers that should be removed. This also could get expended to remove entire use cases.